### DISCUSSION TITLE
Which one is better Dynamic routes or Query string?

### DISCUSSION THREAD

**Discussion by abreujpedro** [vercel/next.js#29102] [Help] [answered]
There's a page that I need to create, and I want to know when I should use Dynamic Route(using Next) or Query string.

**Comment by wilsonneto-swe** [comment, marked as answer]
It depends, it will be more about semantic then about the Next, because this will be the same question for any framework...

The query string parameters are better for filters or for some small parameters for the page. (paginations, filters, a highlight in one page)

The dynamic route is better for pages, a movie page: "/details/wonder-woman"...

A simple example "/catalog/sony?page=1", page about sony catalog on page 1.

But stay aware, next.js doesn't cache considering the query string yet, only considers dynamic routes, then if you will work with cache the dynamic route will be better.


I hope it helps you man!
All the best in your projects.
Best regards.

**Reply by abreujpedro** [reply]
I decided to use dynamic routes, I think that next has a problem with query string in some ways, thanks for help.
